nipype/interfaces/image.py

```diff
@@ -9,6 +9,84 @@
 from .base import (SimpleInterface, TraitedSpec, BaseInterfaceInputSpec,
                    traits, File)
 
+
+class RescaleInputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, mandatory=True,
+                   desc='Skull-stripped image to rescale')
+    ref_file = File(exists=True, mandatory=True,
+                    desc='Skull-stripped reference image')
+    invert = traits.Bool(desc='Invert contrast of rescaled image')
+    percentile = traits.Range(low=0., high=50., value=0., usedefault=True,
```

Comment:
Hmm. I expected that people would probably only ever want 0 or 1 (but left room for tweaking), but I can add more flexibility than this. This leads me to wonder if we want to permit different percentiles for the input and reference images. In which case we need 4 values, not 2.

WDYT? If possible, it'd be nice to allow a single knob, to keep it simple until somebody needs more refined control.

Reply:
WDYT?:

```Python
    percentile = traits.Either(
        traits.Range(low=0., high=50, value=0),  # One-knob behavior
        traits.Tuple(
            traits.Either(None, traits.Range(low=0., high=50, value=0)),
            traits.Either(None, traits.Range(low=50., high=100, value=100)),
        ), desc='...')
```

Then, if it is not a tuple, it has your one-knob mirror behavior. Otherwise you have a tuple, and you can always set one of the boundaries.

Reply:
My fixation for this is that T1 images typically show background pixels under the 15% and, except when there are WM hyperintensities, you probably want 98% or higher for the upper limit.

Reply:
So, if I'm reading this correct, we don't need to be concerned with supporting different percentiles for the input and the reference?

> My fixation for this is that T1 images typically show background pixels under the 15% and, except when there are WM hyperintensities, you probably want 98% or higher for the upper limit.

This is true for skull-stripped images?

Reply:
Nope, but I think we should aim for this interface to work well in any situation.

Reply:
> I think we should aim for this interface to work well in any situation.

I don't think this interface will work at all well in non-skull-stripped cases. Calculating the magic numbers needed won't be significantly easier or more robust than even mediocre skull stripping, if I were to make a guess.

> I guess we should specify that this interface expects the reference file to be "consistent" (as in already thresholded).

I do say:

```Python
in_file = File(exists=True, mandatory=True,
               desc='Skull-stripped image to rescale')
ref_file = File(exists=True, mandatory=True,
                desc='Skull-stripped reference image')
```

And:

```
Rescales the non-zero portion of ``in_file`` to match the bounds of the
non-zero portion of ``ref_file``.
```

But I can very explicitly say in the docstring that they should be skull-stripped. And would you suggest not applying the supplied percentiles to the reference image, as well?

I guess what's your vision for what this interface should do when percentiles != (0, 100)?